daemon/daemon.go

```diff
@@ -448,6 +448,11 @@ func (daemon *Daemon) registerLink(parent, child *container.Container, alias str
 
 // SetClusterProvider sets a component for querying the current cluster state.
 func (daemon *Daemon) SetClusterProvider(clusterProvider cluster.Provider) {
+	// Daemon is in charge of removing the attachable networks with
+	// connected containers when the node leaves the swarm
+	if clusterProvider == nil {
+		daemon.clearAttachableNetworks()
+	}
```

Comment:
I see your point, and I see a different solution.

Let's me summarize: 

From daemon POV, `SetClusterProvider()` has the current semantic: "The node has joined the swarm/ The node has left the swarm" (based on the value of the passed provider pointer).

The cleaning up the attachable networks which are being kept alive by user run containers is to be done by the daemon when it gets notified the node left the swarm. 

I am proposing we could rename the API from its current technical meaning to one which carries the higher picture context:
`SetClusterProvider(c != nil)` => `DaemonJoinedCluster(c)`
and 
`SetClusterProvider(c == nil)` => `DaemonLeftCluster()`

I think this would make it natural to the reader when he sees `DaemonLeftCluster()` invoke the nw cleanup. WDYT ?